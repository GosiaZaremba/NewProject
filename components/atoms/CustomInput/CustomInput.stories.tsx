import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { CustomInput } from './CustomInput';
import { boolean, text } from '@storybook/addon-knobs';

storiesOf('atoms/CustomInput', module)
    .addDecorator((story) => story())
    .add('default', () => (
        <CustomInput
            placeholder={text('placeholder', 'Email')}
            secureTextEntry={boolean('secureTextEntry', true)}
        />
    ));
