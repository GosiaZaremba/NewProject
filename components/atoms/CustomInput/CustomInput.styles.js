import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  input: {
    borderBottomWidth: 1,
    borderBottomColor: Colors.five,
    padding: 10,
  },
});
