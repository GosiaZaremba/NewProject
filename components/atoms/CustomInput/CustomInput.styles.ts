import {StyleSheet} from 'react-native';
import {Colors} from '../../../constants/colors';

export default StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: Colors.five,
    padding: 10,
    width: '100%',
    borderRadius: 5,
  },
});
