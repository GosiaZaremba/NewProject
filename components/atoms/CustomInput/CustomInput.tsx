import React, {
    forwardRef,
    useImperativeHandle,
    useRef,
    useState,
} from 'react';
import { TextInput } from 'react-native';
import styles from './CustomInput.styles';

export type Props = {
    placeholder: string;
    secureTextEntry: boolean;
    testID?: string;
};

export type CustomInputReference = {
    getValue: () => string;
    focus: () => void;
    // isFocused: () => boolean | undefined;
};

const CustomInputWithReference: React.ForwardRefRenderFunction<
    CustomInputReference,
    Props
> = ({ placeholder, secureTextEntry, testID }, ref) => {
    const [value, setValue] = useState<string>('');

    const inputReference = useRef<TextInput>(null);

    const onChangeText = (val: string) => {
        setValue(val);
    };

    useImperativeHandle(ref, () => ({
        getValue: () => {
            return value;
        },
        focus: () => {
            inputReference.current?.focus();
        },
        // isFocused: () => {
        //     return inputReference.current?.isFocused();
        // },
    }));

    return (
        <TextInput
            onChangeText={onChangeText}
            placeholder={placeholder}
            secureTextEntry={secureTextEntry}
            value={value}
            style={styles.input}
            ref={inputReference}
            testID={testID}
        />
    );
};

export const CustomInput = forwardRef(CustomInputWithReference);
