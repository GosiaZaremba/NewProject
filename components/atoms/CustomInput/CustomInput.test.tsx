import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { CustomInput, CustomInputReference } from './CustomInput';

describe('Atom Input', () => {
    test('Component Snapshot', () => {
        const tree = renderer
            .create(
                <CustomInput
                    placeholder={'Custom Input'}
                    secureTextEntry={false}
                    testID={'atom-input'}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure variants works correctly', () => {
        const { getByTestId } = render(
            <CustomInput
                placeholder={'Custom Input'}
                secureTextEntry={false}
                testID={'atom-input'}
            />
        );
        const customInput = getByTestId('atom-input');
        fireEvent.changeText(customInput, '123');
        expect(customInput.props.value).toBe('123');
        expect(customInput.props.placeholder).toBe('Custom Input');
        expect(customInput.props.secureTextEntry).toBe(false);
    });

    test('ensure reference is passed', () => {
        const inputRef = React.createRef<CustomInputReference>();
        const { getByTestId } = render(
            <CustomInput
                placeholder={'Custom Input'}
                secureTextEntry={false}
                testID={'atom-input'}
                ref={inputRef}
            />
        );
        const customInput = getByTestId('atom-input');
        fireEvent.changeText(customInput, 'input content');
        expect(inputRef.current?.getValue()).toBe('input content');
        fireEvent(customInput, 'inputRef.current?.focus()');
        // fireEvent(customInput, 'inputRef.current?.isFocused()');
        inputRef.current?.focus();
        // expect(inputRef.current?.focus()).toBeUndefined();
        // console.log(inputRef.current?.isFocused());
    });
});
