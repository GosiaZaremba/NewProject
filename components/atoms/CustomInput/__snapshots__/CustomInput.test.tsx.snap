// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Atom Component Snapshot 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="Custom Input"
  secureTextEntry={false}
  style={
    Object {
      "borderColor": "#908D8E",
      "borderRadius": 5,
      "borderWidth": 1,
      "padding": 10,
      "width": "100%",
    }
  }
  testID="atom-input"
  value=""
/>
`;
